falcon/util/uri.py

```diff
@@ -292,9 +292,9 @@ def parse_query_string(query_string, keep_blank_qs_values=False):
             # a list, or append the next value to the list.
             old_value = params[k]
             if isinstance(old_value, list):
-                old_value.append(v)
+                old_value.append(decode(v))
             else:
-                params[k] = [old_value, v]
+                params[k] = [old_value, decode(v)]
 
         else:
             if ',' in v:
```

Comment:
Seems like v must be decoded before checking if it contains commas, since the commas might be encoded. What do you think about factoring out the `decode(v)` and placing it above L290, e.g.:

```python
v = decode(v)

for k in params:
    # ...
```